onnxruntime/core/xnnpack/schema/xnnpack_onnx_defs.h

```diff
@@ -0,0 +1,119 @@
+                              uint32_t dilation_w, ONNXRUNTIME_XNNPACK_PADDING_MODE padding_mode, T2& output) {
+  if (in_channels != in_channels1) {
+    return OnnxStatus(::ONNX_NAMESPACE::Common::StatusCategory::NONE, ::ONNX_NAMESPACE::Common::StatusCode::FAIL);
+  }
+
+  SetDimValue(output[0], batch_shape);
+  ptrdiff_t output1, output2;
+  if (padding_mode == ONNXRUNTIME_XNNPACK_PADDING_MODE_SAME) {
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeSame(in_height, strides_h, &output1));
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeSame(in_width, strides_w, &output2));
+  } else if (padding_mode == ONNXRUNTIME_XNNPACK_PADDING_MODE_VALID) {
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeValid(in_height, strides_h, filter_height, dilation_h, &output1));
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeValid(in_width, strides_w, filter_width, dilation_w, &output2));
+  } else {
+    return OnnxStatus(::ONNX_NAMESPACE::Common::StatusCategory::NONE, ::ONNX_NAMESPACE::Common::StatusCode::FAIL,
+                      "Invalid padding mode.");
+  }
+
+  SetDimValue(output[3], out_channels);
+  SetDimValue(output[1], output1);
+  SetDimValue(output[2], output2);
+  return ::ONNX_NAMESPACE::Common::Status::OK();
+}
+template <typename T1, typename T2>
+OnnxStatus MaxPoolShapeInference(const T1& batch_shape, ptrdiff_t in_height, ptrdiff_t in_width, ptrdiff_t in_channels,
+                                 ptrdiff_t pooling_height, ptrdiff_t pooling_width, uint32_t strides_h,
+                                 uint32_t strides_w, uint32_t dilation_h, uint32_t dilation_w,
+                                 ONNXRUNTIME_XNNPACK_PADDING_MODE padding_mode, T2& output) {
+  SetDimValue(output[0], batch_shape);
+  ptrdiff_t output1, output2;
+  if (padding_mode == ONNXRUNTIME_XNNPACK_PADDING_MODE_SAME) {
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeSame(in_height, strides_h, &output1));
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeSame(in_width, strides_w, &output2));
+  } else if (padding_mode == ONNXRUNTIME_XNNPACK_PADDING_MODE_VALID) {
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeValid(in_height, strides_h, pooling_height, dilation_h, &output1));
+    ONNX_RETURN_IF_ERROR(ComputeOutputSizeValid(in_width, strides_w, pooling_width, dilation_w, &output2));
+  } else {
+    return OnnxStatus(::ONNX_NAMESPACE::Common::StatusCategory::NONE, ::ONNX_NAMESPACE::Common::StatusCode::FAIL,
+                      "Invalid padding mode.");
+  }
+
+  // The number of output channels is always equals to the number of input channels
+  SetDimValue(output[3], in_channels);
+  SetDimValue(output[1], output1);
+  SetDimValue(output[2], output2);
+  return ::ONNX_NAMESPACE::Common::Status::OK();
+}
+
+}  // namespace xnnpack
+}  // namespace onnxruntime
+
+#define ONNX_XNNPACK_OPERATOR_SET_SCHEMA(name, ver, impl) \
+  ONNX_OPERATOR_SET_SCHEMA_EX(name, XnnPack, ::onnxruntime::kXNNPackDomain, ver, true, impl)
```

Comment:
>kXNNPackDomain [](http://example.com/codeflow?start=60&length=14)

Mreaning for older onnx models to benefit from XNN, they will have to be reexported to have xnn ops?
Why the EP approach is not preferred here?

Reply:
I had a discussion with Randy and Pranav about adding XNNPack as an EP vs contrib/custom ops. I think both are fine. I don't exclude the possibility of making it an EP. 

> they will have to be reexported to have xnn ops?

No. I added a graph transformer in ORT for it.